### PR TITLE
adding tiny-js-world task

### PR DESCRIPTION
**Pull request by 5Mountains**
### a tiny js world

[Demo](https://5mountains.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/5Mountains/frontend-2021-homeworks/tree/tiny-js-world/submissions/5mountains/tiny-js-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/5mountains/tiny-js-world/index.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+/* Print examples:
+   print('ABC');
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+let residents = new Array(man, woman, cat, dog, catWoman);
```
Comment (madmaxWMFU) [inline submissions/5mountains/tiny-js-world/index.js]: The variable `residents` must be `const`

You can also create an array using the syntax `[a, b, c]`.

File: submissions/5mountains/tiny-js-world/index.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+for(let character of residents) {
```
Comment (madmaxWMFU) [inline submissions/5mountains/tiny-js-world/index.js]: You can also try the method `forEach`.

File: submissions/5mountains/tiny-js-world/index.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+   let sentence = `Welcome, let me tell you briefly about our resident - a wonderful <b>${character.species}</b> creature, the name is <b>${character.name}</b>! This member is the owner of legs, number is ${character.legs}, of hands, number is ${character.hands} or of paws, number is ${character.paws}, usually the greeting is <i style="text-decoration:underline;">${character.saying}</i> and friends of this inhabitant are ${character.friends.join(', ')}.`;
```
Comment (madmaxWMFU) [inline submissions/5mountains/tiny-js-world/index.js]: We have a pattern string here.
Please implement a new function that receives your object and returns a string.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Good job.
See the comments below.

File: submissions/5mountains/tiny-js-world/index.js (hunk 1 above)
Comment (5Mountains) [reply submissions/5mountains/tiny-js-world/index.js]: you check the old version of code) I dit renaming from let to const... Why I cannot create an array using this method? Is it a bad approach?

File: submissions/5mountains/tiny-js-world/index.js (hunk 3 above)
Comment (5Mountains) [reply submissions/5mountains/tiny-js-world/index.js]: Could you please check my latest code version?) Is everything okay there?)

File: submissions/5mountains/tiny-js-world/index.js (hunk 2 above)
Comment (5Mountains) [reply submissions/5mountains/tiny-js-world/index.js]: I'm familiar with this method also, but I decided to use `for...of`. Is this a bad method nowadays, and I always should use `forEach`?

File: submissions/5mountains/tiny-js-world/index.js (hunk 1 above)
Comment (madmaxWMFU) [reply submissions/5mountains/tiny-js-world/index.js]: The syntax is an alternative)

File: submissions/5mountains/tiny-js-world/index.js (hunk 2 above)
Comment (madmaxWMFU) [reply submissions/5mountains/tiny-js-world/index.js]: It makes code more concise.

Comment (5Mountains) [reply submissions/5mountains/tiny-js-world/index.js]: What do you say if I use this method:
```
const printStory = function() {
   let story = '';
   Array.from([man, woman, cat, dog, catWoman], character => {
      result += `HUGE ELEMENT`
   });
   print(story, 'div');
};
```

File: submissions/5mountains/tiny-js-world/index.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+const residents = new Array(man, woman, cat, dog, catWoman);
+
+const createStory = function() {
```
Comment (madmaxWMFU) [inline submissions/5mountains/tiny-js-world/index.js:80]: Use arrow function

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

File: submissions/5mountains/tiny-js-world/index.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+   catWoman = {};
+
+man.species ='human';
+man.name = 'Tom';
+man.gender = 'male';
+man.legs = 2;
+man.hands = 2;
+man.paws = 0;
+man.saying = 'Hello Jenny!';
+man.friends = ['Tom', 'Jenny', 'Rex', 'Selina'];
```
Comment (OleksiyRudenko) [inline submissions/5mountains/tiny-js-world/index.js:23]: The above fragment might look like the following:
```
const man = {
    species: 'human',
    name: 'Tom',
    gender: 'male',
    // ... etc
  },
  woman = {
    species: 'human',
    // ... etc
  },
  // ... etc
```

File: submissions/5mountains/tiny-js-world/index.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+cat.friends = ['Tom', 'Jenny', 'Selina'];
+
+catWoman.species = undefined;
+catWoman.name = 'Selina';
+catWoman.gender = 'female';
+catWoman.legs = undefined;
+catWoman.hands = undefined;
+catWoman.paws = undefined;
+catWoman.saying = cat.saying;
+catWoman.friends = ['Tom', 'Felix'];
```
Comment (OleksiyRudenko) [inline submissions/5mountains/tiny-js-world/index.js:59]: Having this definition for the catWoman the output looks as follows.
image
Doesn't look like what you intended to have.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@5Mountains nice job!
However a couple of improvements are rather required.
One for the sake of code readability, yet another concerns output that would make sense.
Please check the comments below for the details.

File: submissions/5mountains/tiny-js-world/index.js (hunk 5 above)
Comment (5Mountains) [reply submissions/5mountains/tiny-js-world/index.js]: done

File: submissions/5mountains/tiny-js-world/index.js (hunk 6 above)
Comment (5Mountains) [reply submissions/5mountains/tiny-js-world/index.js]: Fixed! Please check the output now (:

File: submissions/5mountains/tiny-js-world/index.js (hunk 4 above)
Comment (5Mountains) [reply submissions/5mountains/tiny-js-world/index.js]: done

File: submissions/5mountains/tiny-js-world/index.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+const createStory = (character, index) => {
+   let speciesDiff;
+   if(character.paws > 0) speciesDiff = `<b>${character.paws}</b> paws`;
+   else speciesDiff = `<b>${character.legs}</b> legs and is <b>${character.hands}</b> hands`;
+   return  `<b>Welcome</b>, wanderer, let me tell you briefly about ${index+1}th our resident - a wonderful <b>${character.species}</b> creature, the name is <b>${character.name}</b>! This member is the owner of ${speciesDiff}, usually the greeting is <i style="text-decoration:underline;"><b>${character.saying}</b></i> and friends of this inhabitant are <b>${character.friends.join(', ')}</b>.`;
+};
+
+Array.from([man, woman, cat, dog, catWoman], (obj, index) => print(createStory(obj, index), 'p'));
```
Comment (madmaxWMFU) [inline submissions/5mountains/tiny-js-world/index.js:85]: In my opinion, the method `map` is better.

Comment (5Mountains) [reply submissions/5mountains/tiny-js-world/index.js:85]: `Array.from()` method includes `map` under the hood, but maybe this kind of writing more difficult to read... I used this approach in order to show that I familiar not only with `map` :) but, if I have to, I car rewrite this!

**Comment by 5Mountains** [comment]
Reducing the output of text content (:

File: submissions/5mountains/tiny-js-world/index.js (hunk 7 above)
Comment (madmaxWMFU) [reply submissions/5mountains/tiny-js-world/index.js]: This would be better readable and more abbreviated)

Comment (5Mountains) [reply submissions/5mountains/tiny-js-world/index.js]: Easily! Refactoring - done (:

**Review by madmaxWMFU — APPROVED** [review]

**Review by OleksiyRudenko — APPROVED** [review]
LGTM 👍

**Pull request merged**